File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int xoracc = 0;

        for(const int& e : nums) xoracc ^= e;

        vector<int> ans{};
        for(auto it = nums.crbegin(); it != nums.crend(); ++it){
            ans.push_back(xoracc ^ ((1 << maximumBit) - 1));
            xoracc ^= *it;
        }

        return ans;
    }
};

File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.erl
-spec impl(Nums :: [integer()], XorAcc :: integer(), MaximumBit :: integer()) -> [integer()].
impl([], _, _) -> [];
impl([H | T], XorAcc, MaximumBit) ->
    [XorAcc bxor ((1 bsl MaximumBit) - 1)] ++ impl(T, XorAcc bxor H, MaximumBit).

-spec get_maximum_xor(Nums :: [integer()], MaximumBit :: integer()) -> [integer()].
get_maximum_xor(Nums, MaximumBit) ->
    XorAcc = lists:foldl(
        fun (X, Acc) ->
            Acc bxor X
        end, 0, Nums
    ),
    impl(lists:reverse(Nums), XorAcc, MaximumBit).